### PR TITLE
[GPU]get data type of conv weights from node.weights() when network is internal

### PR DESCRIPTION
**Pull request by wilson-seok**
### Details:
 - Convolution filter from specific model has input_layout when network is internal. This causes seg fault during checking data type between input and weights in validate_impl() after implementing https://github.com/openvinotoolkit/openvino/pull/6897.
 - To avoid the issue, I added condition to check network.is_internal() and get data type from node.weights() instead of weights_memory(0).

### Tickets:
 - 87438

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/impls/ocl/convolution.cpp [hunk 1]
```diff
@@ -31,13 +31,19 @@ struct convolution_impl : typed_primitive_impl_ocl<convolution> {
 
         auto outer_id = _outer.id();
         auto data_type = instance.node.input().get_output_layout().data_type;
+        cldnn::data_types weights_data_type;
+        if (instance.get_network().is_internal()) {
+           weights_data_type = instance.node.weights().get_output_layout().data_type;
+        } else {
+           weights_data_type = instance.weights_memory(0)->get_layout().data_type;
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/src/graph/impls/ocl/convolution.cpp:38]: Is this branch really needed? Won't the following work?
`cldnn::data_types weights_data_type = instance.node.weights().get_output_layout().data_type;`

File: src/plugins/intel_gpu/src/graph/impls/ocl/convolution.cpp [hunk 2]
```diff
@@ -31,13 +31,19 @@ struct convolution_impl : typed_primitive_impl_ocl<convolution> {
...
+        }
 
         // Integer signed/unsigned is ok for convoluiton
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/src/graph/impls/ocl/convolution.cpp:41]: please create unit tests for that case

File: src/plugins/intel_gpu/src/graph/impls/ocl/convolution.cpp (hunk 1 above)
Comment (yeonbok) [reply src/plugins/intel_gpu/src/graph/impls/ocl/convolution.cpp:38]: I think this is good too. Just made this branch to preseve original path too.. :)

Comment (wilson-seok) [reply src/plugins/intel_gpu/src/graph/impls/ocl/convolution.cpp]: Done. Thanks!

File: src/plugins/intel_gpu/src/graph/impls/ocl/convolution.cpp (hunk 2 above)
Comment (wilson-seok) [reply src/plugins/intel_gpu/src/graph/impls/ocl/convolution.cpp]: @vladimir-paramuzov So unit test for getting instance.node.weights().get_output_layout().data_type? Do we have similar test case in clDNN unit test?

Comment (wilson-seok) [reply src/plugins/intel_gpu/src/graph/impls/ocl/convolution.cpp]: @vladimir-paramuzov Convolution unit test already includes the case of weights located in input_layout primitive. https://github.com/wilson-seok/openvino/blob/749f15cf8b7c9db5a50b5bda4c458f883412e529/src/plugins/intel_gpu/tests/test_cases/convolution_gpu_test.cpp#L1298
Do you mean this kind of test or another one?

Comment (vladimir-paramuzov) [reply src/plugins/intel_gpu/src/graph/impls/ocl/convolution.cpp]: Yes, something like that, but new test case should cover fixed scenario, i.e. it must fail w/o your fix.

Comment (wilson-seok) [reply src/plugins/intel_gpu/src/graph/impls/ocl/convolution.cpp]: @vladimir-paramuzov Added unit test for the case. Thanks for the comments!

**Review by vladimir-paramuzov — APPROVED** [review]

**Pull request merged**